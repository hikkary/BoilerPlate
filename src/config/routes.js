import React from 'react'
import { Route, BrowserRouter as Router, Switch } from 'react-router-dom'

import Login from '../containers/login'
import HomePage from '../containers/homePage'
import HouseCharacters from '../containers/houseCharacters'
import Ranking from '../containers/ranking'
import Options from '../containers/options'

export default class Routes extends React.Component {
  render() {
    return (
      <Router>
        <Switch>
          <Route exact path="/" component={Login} />
          <Route path="/home" component={HomePage} />
          <Route path="/house/:houseName" component={HouseCharacters} />
          <Route path="/ranking" component={Ranking} />
          <Route path="/options" component={Options} />
        </Switch>
      </Router>
    )
  }
}
