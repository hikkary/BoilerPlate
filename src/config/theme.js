export default {
  color: {
    black: '#222',
    white: '#fff',
    loginBackground: '#f2debd',
    bookSeparator: '#580303',
    primaryColor: '#7f0909',
    secondaryColor: '#940B0B',
    tertiaryColor: '#CB4B4B',
    gray: '#979797'
  }
}
