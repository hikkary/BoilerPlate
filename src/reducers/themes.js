import { CHANGE_THEME } from '../actions/themes'
import { themeLight } from '../config/themes'

const initialState = {
  currentTheme: themeLight
}

export default (state = initialState, action) => {
  switch (action.type) {
    case CHANGE_THEME:
      return {
        ...state,
        currentTheme: action.theme
      }

    default:
      return state
  }
}
