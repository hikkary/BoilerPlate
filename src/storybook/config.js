import React from 'react'
import { addDecorator, configure } from '@storybook/react'
import { ThemeProvider } from 'styled-components'
import centered from '@storybook/addon-centered'
import '../../node_modules/react-vis/dist/style.css'

import theme from '../config/theme'

const ThemeDecorator = storyFn => (
  <ThemeProvider theme={theme}>{storyFn()}</ThemeProvider>
)
addDecorator(ThemeDecorator)
addDecorator(centered)

function loadStories() {
  require('../components/input/story.js')
  require('../components/loginForm/story.js')
  require('../components/panel/story.js')
  require('../components/texts/story.js')
  require('../components/graphTest/story.js')
  // You can require as many stories as you need.
}

configure(loadStories, module)
