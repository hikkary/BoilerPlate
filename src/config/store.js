import { applyMiddleware, createStore } from 'redux'
import logger from 'redux-logger'
// used for async func use on redux
// import thunk from 'redux-thunk'

import reducers from '../reducers'

const store = createStore(reducers, applyMiddleware(logger))

export default store
