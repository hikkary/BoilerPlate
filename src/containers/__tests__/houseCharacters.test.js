import React from 'react'
import MockAdapter from 'axios-mock-adapter'

import { shallow, mount } from 'enzyme'
import axios from 'axios'
import { ThemeProvider } from 'styled-components'
import { MemoryRouter } from 'react-router-dom'

import ExerciceSeven from '../houseCharacters'
import { themeLight } from '../../config/themes'
import 'jest-styled-components'

const mockData = { results: [{ name: 'harry Potter' }] }

const mock = new MockAdapter(axios)
mock
  .onGet('http://hp-api.herokuapp.com/api/characters/house/gryffindor')
  .reply(200, mockData)

let mockingMatch = {
  params: {
    houseName: 'gryffindor'
  }
}

class ThemeWrapper extends React.Component {
  render() {
    return (
      <ThemeProvider theme={themeLight}>
        <MemoryRouter>{this.props.children}</MemoryRouter>
      </ThemeProvider>
    )
  }
}

it('render properly', () => {
  const component = shallow(
    <ThemeWrapper>
      <ExerciceSeven match={mockingMatch} />
    </ThemeWrapper>
  )

  expect(component).toMatchSnapshot()
  component.unmount()
})

// it('should have one element', () => {
//   const component = shallow(
//     <ThemeWrapper>
//       <ExerciceSeven match={mockingMatch} />
//     </ThemeWrapper>
//   ).setState({ characters: mockData.results })
//   // .update()
//   // component
//   // component.update()
//   console.log(component.state())

//   const p = component.fuind('p')
//   console.log(p.length)
//   console.log(component.find('[className="characterName"]').html())
//   // expect(component).toMatchSnapshot()
// })
