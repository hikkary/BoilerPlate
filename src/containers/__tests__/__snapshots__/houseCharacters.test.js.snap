// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render properly 1`] = `
<ThemeProvider
  theme={
    Object {
      "color": Object {
        "backgroundColor": "#fff",
        "black": "#222",
        "bookSeparator": "#580303",
        "gray": "#979797",
        "gryffindorPrimary": "#7f0909",
        "gryffindorSecondary": "#d3a625",
        "hufflepuffPrimary": "#ecb939",
        "hufflepuffSecondary": "#372e29",
        "loginBackground": "#f2debd",
        "primaryColor": "#7f0909",
        "ravenclawPrimary": "#0e1a40",
        "ravenclawSecondary": "#946b2d",
        "secondaryColor": "#940B0B",
        "slytherinPrimary": "#1a472a",
        "slytherinSecondary": "#aaaaaa",
        "tertiaryColor": "#CB4B4B",
        "white": "#fff",
      },
    }
  }
>
  <MemoryRouter>
    <HomePage
      match={
        Object {
          "params": Object {
            "houseName": "gryffindor",
          },
        }
      }
    />
  </MemoryRouter>
</ThemeProvider>
`;
